internal/worker_deployments.go

```diff
@@ -0,0 +1,58 @@
+package internal
+
+import (
+	"fmt"
+	"strings"
+
+	deploymentpb "go.temporal.io/api/deployment/v1"
+)
+
+// Represents the version of a specific worker deployment.
+//
+// NOTE: Experimental
+//
+// Exposed as: [go.temporal.io/sdk/client.WorkerDeploymentVersion], [go.temporal.io/sdk/worker.WorkerDeploymentVersion]
+type WorkerDeploymentVersion struct {
+	// The name of the deployment this worker version belongs to
+	DeploymentName string
+	// The build id specific to this worker
+	BuildId string
+}
+
+func workerDeploymentVersionFromProto(wd *deploymentpb.WorkerDeploymentVersion) WorkerDeploymentVersion {
+	return WorkerDeploymentVersion{
+		DeploymentName: wd.DeploymentName,
+		BuildId:        wd.BuildId,
+	}
+}
+
+func (wd *WorkerDeploymentVersion) toProto() *deploymentpb.WorkerDeploymentVersion {
+	return &deploymentpb.WorkerDeploymentVersion{
+		DeploymentName: wd.DeploymentName,
+		BuildId:        wd.BuildId,
+	}
+}
+
+func (wd *WorkerDeploymentVersion) ToCanonicalString() string {
```

Comment:
should this be also private the same way that `workerDeploymentVersionFromString` is? it is only useful for the deprecated fields, right?